graph2/graph.proto

```diff
@@ -0,0 +1,307 @@
+syntax = "proto3";
+package graph2;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "sourcegraph.com/sqs/pbtypes/timestamp.proto";
+import "sourcegraph.com/sourcegraph/srclib/toolref.proto";
+
+option (gogoproto.goproto_getters_all) = false;
+
+// GitHubRepo holds additional metadata about GitHub repos.
+message GitHubRepo {
+	int32 stars = 1;
+}
+
+// RepoPermissions describes the possible permissions that a user (or an anonymous
+// user) can be granted to a repository.
+message RepoPermissions {
+	bool read = 1;
+	bool write = 2;
+	bool admin = 3;
+}
+
+message TreeKey {
+  // TreeType is the type of source tree.
+  // (git, hg, build-system specific like pip or npm, ftp)
+  string tree_type = 1;
```

Comment:
Minor. Leading whitespace is inconsistent. It's a tab above in `GitHubRepo` and `RepoPermissions`, but 2 spaces here and below. It would be better to keep a consistent style. I'd be in favor of staying close to gofmt and using a single tab for indentation (but spaces for alignment).